### PR TITLE
Add behavior of the HDMA HBlank mode in halt mode

### PR DESCRIPTION
**Pull request by alloncm**

### CODE REVIEW

**Review by ISSOtm — APPROVED** [review]
I have seen this confirmed several times, so I'm approving this despite the lack of a reference. (Better have the information *somewhere*!)

Thank you!

**Comment by alloncm** [comment]
I can add a reference to a test ROM I wrote. 
The issue with it is that it doesn't verify the transfer continues once the CPU is unhalted (that's why I didn't add it in this PR). 
I verified this behavior manually with SameBoy though.

https://github.com/alloncm/MagenTests?tab=readme-ov-file#vram-dma-hblank-mode

**Review by avivace — APPROVED** [review]
OK for me in this state but I'd add the link to the test rom

File: src/CGB_Registers.md [hunk 1]
```diff
@@ -79,7 +79,10 @@ the program execution continues during the "spaces" between each data
 block. Note that the program should not change the Destination VRAM bank
 (FF4F), or the Source ROM/RAM bank (in case data is transferred from
 bankable memory) until the transfer has completed! (The transfer should
-be paused as described below while the banks are switched)
+be paused as described below while the banks are switched).
+
+Upon halting the CPU (using the [halt instruction](<#Using the HALT Instruction>)),
+the transfer will also be halted and will be resumed only when the CPU resumes execution.
```
Comment (avivace) [inline src/CGB_Registers.md:85]:
```suggestion
the transfer will also be halted and will be resumed only when the CPU resumes execution ([test rom](https://github.com/alloncm/MagenTests?tab=readme-ov-file#vram-dma-hblank-mode) exhibiting this behaviour)
```

**Review by avivace — CHANGES_REQUESTED** [review]
e.g. like this

**Review by avivace — APPROVED** [review]

**Pull request merged**